azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/PartitionReceiver.java

```diff
@@ -288,7 +305,28 @@ public final long getEpoch()
 			@Override
 			public Iterable<EventData> apply(Collection<Message> amqpMessages)
 			{
-				return EventDataUtil.toEventDataCollection(amqpMessages);
+                                PassByRef<Message> lastMessageRef = null;
+                                if (PartitionReceiver.this.receiverOptions != null && PartitionReceiver.this.receiverOptions.getReceiverRuntimeMetricEnabled())
+                                   lastMessageRef = new PassByRef<>();
+                                
+				Iterable<EventData> events = EventDataUtil.toEventDataCollection(amqpMessages, lastMessageRef);
+                                
+                                if (lastMessageRef != null && lastMessageRef.get() != null) {
+                                    
+                                    if (PartitionReceiver.this.runtimeInformation == null)
+                                        PartitionReceiver.this.runtimeInformation = new ReceiverRuntimeInformation(PartitionReceiver.this.partitionId);
+                                    
+                                    if (lastMessageRef.get().getDeliveryAnnotations() != null && lastMessageRef.get().getDeliveryAnnotations().getValue() != null) {
+                                        
+                                        Map<Symbol, Object> deliveryAnnotationsMap = lastMessageRef.get().getDeliveryAnnotations().getValue();
```

Comment:
submitted a NEW PR for the feedback: [https://github.com/Azure/azure-event-hubs-java/pull/80](https://github.com/Azure/azure-event-hubs-java/pull/80)

---
In reply to: [106258518](https://github.com/Azure/azure-event-hubs-java/pull/76#discussion_r106258518) [](ancestors = 106258518)